kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQueryHelper.java

```diff
@@ -29,43 +29,64 @@
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
 
 /**
  * Convenience class for creating a default {@link com.google.cloud.bigquery.BigQuery} instance,
  * with or without login credentials.
  */
 public class BigQueryHelper {
   private static final Logger logger = LoggerFactory.getLogger(BigQueryHelper.class);
+  private static String keyFileType;
 
   /**
    * Returns a default {@link BigQuery} instance for the specified project with credentials provided
    * in the specified file, which can then be used for creating, updating, and inserting into tables
    * from specific datasets.
    *
    * @param projectName The name of the BigQuery project to work with
-   * @param keyFilename The name of a file containing a JSON key that can be used to provide
+   * @param keyFile The name of a file containing a JSON key that can be used to provide
    *                    credentials to BigQuery, or null if no authentication should be performed.
    * @return The resulting BigQuery object.
    */
-  public BigQuery connect(String projectName, String keyFilename) {
-    if (keyFilename == null) {
+  public BigQuery connect(String projectName, String keyFile) {
+    if (keyFile == null) {
       return connect(projectName);
     }
-
-    logger.debug("Attempting to open file {} for service account json key", keyFilename);
-    try (InputStream credentialsStream = new FileInputStream(keyFilename)) {
-      logger.debug("Attempting to authenticate with BigQuery using provided json key");
+    //
+    logger.debug("Attempting to open file {} for service account json key", keyFile);
+    InputStream credentialsStream;
+    try {
+      if (keyFileType != null && keyFileType.equals("JSON")) {
+        credentialsStream = new ByteArrayInputStream(keyFile.getBytes(StandardCharsets.UTF_8));
+      } else {
+        credentialsStream = new FileInputStream(keyFile);
+      }
       return new
           BigQueryOptions.DefaultBigQueryFactory().create(
-          BigQueryOptions.newBuilder()
-          .setProjectId(projectName)
-          .setCredentials(GoogleCredentials.fromStream(credentialsStream))
-          .build()
+              BigQueryOptions.newBuilder()
+                  .setProjectId(projectName)
+                  .setCredentials(GoogleCredentials.fromStream(credentialsStream))
+                  .build()
       );
     } catch (IOException err) {
       throw new BigQueryConnectException("Failed to access json key file", err);
     }
   }
+  /**
+   * Returns a default {@link BigQuery} instance for the specified project with credentials provided
+   * in the specified file, which can then be used for creating, updating, and inserting into tables
+   * from specific datasets.
+   *
+   * @param keyFileType The type of keyFile config we can expect. This is either a String
```

Comment:
I am confused. If this is expected to be one of a limited number of values, it may be a good idea to specify which values are valid.